plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -334,7 +384,18 @@ def sweep(self, arguments: List[str]) -> None:
                                 "The number of the values and the number of the objectives are"
                                 f" mismatched. Expect {len(directions)}, but actually {len(values)}."
                             )
-                    study.tell(trial=trial, state=state, values=values)
+                    try:
+                        study.tell(trial=trial, state=state, values=values)
+                    except RuntimeError as e:
+                        if (
+                            is_grid_sampler
+                            and "`Study.stop` is supposed to be invoked inside an objective function or a callback."
+                            in str(e)
```

Comment:
This seems a bit brittle.
Why can't we avoid the RuntimeError here?

Reply:
yea i agree this is not ideal :(

I spent some time looking into this but didn't find a better alternative. My theory is optuna sweeper is not making typical use of the API in that the sweeper is controling the number of trials instead of letting optuna handles it internally. 

@toshihikoyanase while we have you here, any better alternative available than catching+checking the exception?